ts-rs/src/export.rs

```diff
@@ -135,14 +135,27 @@ pub(crate) fn export_to<T: TS + ?Sized + 'static, P: AsRef<Path>>(
     if let Some(parent) = path.as_ref().parent() {
         std::fs::create_dir_all(parent)?;
     }
+
     let lock = FILE_LOCK.lock().unwrap();
     {
         // Manually write to file & call `sync_data`. Otherwise, calling `fs::read(path)`
         // immediately after `T::export()` might result in an empty file.
         use std::io::Write;
-        let mut file = File::create(path)?;
+        let mut file = File::create(&path)?;
         file.write_all(buffer.as_bytes())?;
         file.sync_data()?;
+
+        let relative_path = T::output_path()
+            .ok_or_else(std::any::type_name::<T>)
+            .map_err(ExportError::CannotBeExported)?
+            .to_string_lossy();
+
+        std::fs::OpenOptions::new()
+            .read(true)
+            .append(true)
+            .create(true)
+            .open(default_out_dir().join("ts_rs.meta"))?
+            .write_fmt(format_args!("./{relative_path}\n"))?;
```

Comment:
> It's entirely possible that I'm overthinking this, but is this thread-safe as-is, or is some extra sync operation required here?

I was hoping that doing this inside the mutex lock would make it thread safe, but maybe we need to call `sync_data`? I don't actually know either

Reply:
> For most filesystems, the operating system guarantees that all writes are atomic: no writes get mangled because another process writes at the same time.

I think that means we'd be fine even without the lock as long as the write happens in just one syscall, which I think it does as long as we're not using a `BufWriter`.

Reply:
Ah, so that's why we couldn't get it to break all the way back in #221! Nice